### DISCUSSION TITLE
如何保持SDR不要HDR？

### DISCUSSION THREAD

**Discussion by 404gods** [hooke007/mpv_PlayKit#424] [mpv 通用]
之前问了一个播放HDR全屏后灰白的问题https://github.com/hooke007/MPV_lazy/discussions/422，感谢大佬指出是驱动问题。

我的显示器支持HDR但是平时不开，windows也始终关闭HDR，我觉得现在播放非全屏的HDR视频效果还可以，所以想问问怎么在播放HDR的时候始终保持SDR？不要HDR算了。
图一，是非全屏的时候播放效果，我觉得还可以，噪点可能是相机的问题
![非全屏](https://github.com/hooke007/MPV_lazy/assets/50573302/428ee66e-d254-453c-8968-a3d421a59068)
图二，全屏下很灰白，这应该自动转了HDR的真实效果，因为我的显示器和windows都关了HDR。（若是显示器和windows都开启HDR则不再灰白）
![全屏_1](https://github.com/hooke007/MPV_lazy/assets/50573302/6c8e54aa-279b-4410-811e-59cc00dda02b)

安装的是标准版MPV_LAZY, 
mpv.conf文件里修改了以下几个代码：
vo = gpu-next
gpu-context = winvk
hwdec = no
profile = high-quality
dither-depth = 10
deband = yes
profiles.conf保持默认没修改。

**Comment by hooke007** [comment]
你为什么要用winvk

**Reply by 404gods** [reply]
哦因为之前是A卡，我看wiki上建议就用了
但是我试了gpu-context = d3d11，也是全屏就变灰

**Reply by hooke007** [reply]
运行纯净模式，如果还是不行那帮不了你，说明问题大概率不在配置里

**Reply by 404gods** [reply]
试了纯净模式，全屏和非全屏画面颜色没有变化。但是效果和设置vo = gpu差不多，色彩淡没有vo = gpu-next鲜亮

**Reply by hooke007** [reply]
测试模式里单测 vo = gpu-next

**Reply by 404gods** [reply]
mpv-test.conf里只添加了vo = gpu-next，画面和纯净模式一样，全屏正常色彩淡

**Reply by hooke007** [reply]
那你只能去mpv.conf里逐一去删了，我没有思路去快速定位哪个参数了

**Reply by 404gods** [reply]
好的，谢谢大佬，我慢慢试吧

**Reply by 404gods** [reply]
我发现问题在功能参数里，测试模式不带功能参数则画面淡全屏正常，带上功能参数画面鲜亮全屏灰白，大佬看看我的功能参数哪里不对？
 ########
 # 功能 #
 ########

 idle = yes
#input-ipc-server = mpvpipe
 input-builtin-bindings = no
#pause = yes

 target-colorspace-hint = yes

 hr-seek-framedrop = no
 save-position-on-quit = yes
 watch-later-options = start,vid,aid,sid

 keep-open = yes
#ontop = yes
#title-bar = no
 autofit-smaller = 50%x50%
 keepaspect-window = no
#auto-window-resize = no
 hidpi-window-scale = no

 demuxer-max-bytes = 1024MiB
 demuxer-max-back-bytes = 512MiB
#icc-cache = no
 icc-cache-dir = "~~/_cache/icc"
#gpu-shader-cache = no
 gpu-shader-cache-dir = "~~/_cache/shader"
 watch-later-dir = "~~/_cache/watch_later"

======================================================
target-colorspace-hint = yes 这一条注释掉后画面变淡全屏正常

**Reply by hooke007** [reply]
> target-colorspace-hint = yes

这是你自己开的不是我开的

**Reply by 404gods** [reply]
关闭这个并不能得到我要的结果，我只是希望全屏正常画面颜色保持不变淡，算了，这可能没有结果。看了一些国外的讨论，应该是通病。

**Comment by AaronZLT** [comment]
现在解决了吗

**Reply by hooke007** [reply]
我上周找了amd显卡的机器也不能复现，你首先确定你的问题和楼主的是不是一个问题

禁用 target-colorspace-hint
锁定色域输出 https://mpv.io/manual/master/#options-d3d11-output-csp
